### PR TITLE
OCPBUGS-7310: Fix test checking if Machine addresses match VMs

### PR DESCRIPTION
**Pull request by dulek**
The aforementioned test blindly tries to check all the Machines in the cluster, assuming each have a VM in OpenStack. That might not be true as in other tests we create Machines that are supposed to fail provisioning and never produce a VM. This commit fixes this by adding several precautions:

* We only consider machines having addresses set in .status.
* We skip machines without openstack-resourceId annotation.
* We skip checking machines that do not have VM in OpenStack (could have been deleted between we fetched the Machines and did OpenStack query).

### CODE REVIEW

**Comment by rlobillo** [comment]
Hi @dulek,

I think we should not skip machines without openstack-resourceId annotation. That is one of the main checks in the test. 


Isnt "only consider machines having addresses set in .status" enough for stabilizing the test?

**Comment by dulek** [comment]
> Hi @dulek,
> 
> I think we should not skip machines without openstack-resourceId annotation. That is one of the main checks in the test.
> 
> Isnt "only consider machines having addresses set in .status" enough for stabilizing the test?

But if `openstackResourceId == ""`, then the try to fetch the VM from Nova will imminently fail.

**Comment by rlobillo** [comment]
> > Hi @dulek,
> > I think we should not skip machines without openstack-resourceId annotation. That is one of the main checks in the test.
> > Isnt "only consider machines having addresses set in .status" enough for stabilizing the test?
> 
> But if `openstackResourceId == ""`, then the try to fetch the VM from Nova will imminently fail.

Please ignore my comment. The openstack-resourceId annotation is checked on the previous test https://github.com/openshift/openstack-test/blob/85a5fb21736881854d30ef3dfd96daef760806de/test/extended/openstack/servers.go#L88

Sorry for the noise

File: test/extended/openstack/servers.go [hunk 1]
```diff
@@ -119,7 +119,12 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The OpenStack pla
...
-				machineIPs[machineName] = getAddressesFromMachine(machine)
+				machineAddresses := getAddressesFromMachine(machine)
+				machineResourceID := machine.Get("metadata.annotations.openstack-resourceId").String()
+				if len(machineAddresses) == 0 || len(machineResourceID) == 0 {
+					// Only consider machines that have addresses and a VM in OpenStack.
+					continue
+				}
 
 				g.By(fmt.Sprintf("Gather Openstack attributes for machine %q", machineName))
 				instance, err := servers.Get(computeClient, machine.Get("metadata.annotations.openstack-resourceId").String()).Extract()
```
Comment (mandre) [inline test/extended/openstack/servers.go:130]:
```suggestion
				instance, err := servers.Get(computeClient, machineResourceID).Extract()
```

**Review by mandre — COMMENTED** [review]
Just a small nit, otherwise LGTM.

File: test/extended/openstack/servers.go (hunk 1 above)
Comment (dulek) [reply test/extended/openstack/servers.go]: Done!

**Review by mandre — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**